### PR TITLE
Membership event embed

### PR DESCRIPTION
**Pull request by davidrapson**
This PR adds a new element type for enhancing embedded Membership events, based on `rich-links`.
- Markup and styles for enhanced version provided by an endpoint on Membership: https://membership.theguardian.com/event/the-observer-election-countdown-16253236869/card (See https://github.com/guardian/membership-frontend/pull/477)

**Enhanced version**
event-embed-enhanced

**Default appearance**
event-embed-unsupported

This is my first PR so let me know if you need any more information or I've missed some steps.

@mattandrews @phamann

### CODE REVIEW

**Comment by mattandrews** [comment]
LGTM :+1: 
@phamann?

**Comment by davidrapson** [comment]
@sndrs @rich-nguyen got time to review this?

Also, once this has been reviewed it would be good to pair with someone to go through the deployment process.

File: static/src/javascripts/projects/common/modules/article/membership-events.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    $,
+    ajax
+) {
+
+    var ELEMENT_INITIAL_CLASS = 'element-membership--not-upgraded',
+        ELEMENT_UPGRADED_CLASS = 'element-membership--upgraded';
+
+    function upgradeEvent(el) {
+        var href = $('a', el).attr('href'),
+            matches = href.match(/https:\/\/membership.theguardian.com/);
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/article/membership-events.js]: there is a non-regex value for the membership url in our config object, `membershipUrl`, might be safer to use that.

File: static/src/javascripts/projects/common/modules/article/membership-events.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        if (matches) {
+            return ajax({
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/article/membership-events.js]: do you need to return anything at all? Then you wouldn't need to `Promise.resolve(null)` either

**Comment by rich-nguyen** [comment]
Super :+1:

**Comment by rich-nguyen** [comment]
@davidrapson sure, I can go through the deployment process with you

**Pull request merged**